src/controllers/auth.js

```diff
@@ -1,11 +1,92 @@
+const jwt = require('jsonwebtoken');
+
+const bcrypt = require('bcrypt');
+
+const saltRounds = 10;
+
+const User = require('../models/user');
+
 module.exports = {
-  signInUser: (req, res) => {
-    res.send('You successfully signed in');
+  signInUser: async (req, res) => {
+    const { username, email, password, rememberMe } = req.body;
+    let cookieAge = 24 * 3600;
+    try {
+      const user = await User.findOne({
+        $or: [{ email: email }, { username: username }],
+      });
+      /* eslint object-shorthand: ["warn"] */
+      if (!user) throw new Error('Wrong username');
+      const validPassword = await bcrypt.compare(password, user.password_hash);
+      if (!validPassword) throw new Error('Wrong password');
+
+      const payload = {
+        username,
+        email: user.email,
+        firstName: user.firstName,
+        lastName: user.lastName,
+      };
+
+      if (rememberMe) {
+        cookieAge = 14 * 24 * 3600;
+      }
+
+      /* eslint no-underscore-dangle: ["error", { "allow": ["_id"] }] */
+      const token = jwt.sign(payload, process.env.JWT_SECRET, {
+        expiresIn: cookieAge,
+        subject: user._id.toString(),
+      });
+
+      res.cookie('token', token, {
+        maxAge: cookieAge * 1000,
+        httpOnly: true,
+      });
+      res.json({ message: 'Successfully signed in' });
+    } catch (err) {
+      res.status(422).json({ message: err.message ?? err });
+    }
   },
-  signUpUser: (req, res) => {
-    res.send('You successfully signed up');
+  // eslint-disable-next-line consistent-return
+  signUpUser: async (req, res) => {
+    try {
+      const {
+        firstName,
+        lastName,
+        username,
+        email,
+        password,
+        password2,
+        address,
+        acceptTos,
+      } = req.body;
```

Comment:
It would be nice to rename `password2` to `passwordConfirm`.
And `acceptTos` to `acceptTerms`. However, does this field exist in the user model schema?

Reply:
`acceptTerms` doesn't exist on user model schema. Should I add it to schema and assign its type to boolean\ default`true` or String/ default 'on', and make it required? But then I need to check if it is `true` or `on` just then make it possible for user to sign up?

Reply:
Yes add it to the schema as a boolean, but don't put default or required on it.
So that frontend developers can have the option to not use it.